app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt

```diff
@@ -122,23 +130,44 @@ class TicketsViewModel(
                 val code = appliedDiscountCode
                 tickets.forEach { ticket ->
                     var price = ticket.price
+                    totalTaxAmount += (ticket.price * taxRate / 100) * qty[index]
```

Comment:
Why are we not using it.second instead of maintaining an index?

Reply:
it is nothing here. We are fetching the list of tickets and qty stored in an array.

Reply:
Why not use ticketAndQty?

Reply:
I think so. It's quite complex. Will try :+1:

Reply:
It's not. it was created exactly so that index can be avoided

Reply:
ticketAndQty is a list, so the index is required for this also. `ticketAndQty[index].second`

Reply:
No, it is a list of pair. So you iterate over it without the need of index

Reply:
Iterating is already used on a list of tickets which is getting from the database for the ticket price. So for this, two parallel iterators are required.

Reply:
Don't use tickets for iteration, use ticketAndQty, that's why it was created by @anhanh11001

Reply:
Well then ticketAndQty doesn't have the ticket price
```
E/TicketsViewModel: (47, 1, 0.0)
E/TicketsViewModel: (48, 2, 0.0)
```
So here we need to fetch selected tickets to get price. Only donation tickets have price at third place in ticketAndQty.

Reply:
Merging, but open an issue to either use or remove ticketAndQty